go/pkg/cache/clustering/cluster_cache.go

```diff
@@ -0,0 +1,129 @@
+package clustering
+
+import (
+	"context"
+	"time"
+
+	cachev1 "github.com/unkeyed/unkey/go/gen/proto/cache/v1"
+	"github.com/unkeyed/unkey/go/pkg/cache"
+	"github.com/unkeyed/unkey/go/pkg/eventstream"
+	"github.com/unkeyed/unkey/go/pkg/otel/logging"
+)
+
+// ClusterCache wraps a local cache and automatically handles distributed invalidation
+// across cluster nodes using an event stream.
+type ClusterCache[V any] struct {
+	localCache cache.Cache[string, V]
+	topic      *eventstream.Topic[*cachev1.CacheInvalidationEvent]
+	cacheName  string
+	nodeID     string
+	logger     logging.Logger
+}
+
+// Config configures a ClusterCache instance
+type Config[V any] struct {
+	// Local cache instance to wrap
+	LocalCache cache.Cache[string, V]
```

Comment:
Don't we need to allow for K V instead of string V when we use scoped keys? 

Unless we make scoped keys implement a stringer interface?